xpcPlugin/DataManager.cpp

```diff
@@ -792,21 +793,61 @@ namespace XPC
 		Set(DREF_FlapActual, value);
 	}
 
-	void DataManager::Execute(const std::string& comm)
+	void DataManager::MomentaryCommand(const std::string& comm)
 	{
-		Log::FormatLine(LOG_INFO, "DMAN", "Executing command (value:%s)", comm.c_str());
+		Log::FormatLine(LOG_INFO, "DMAN", "Executing momentary command (value:%s)", comm.c_str());
 
  		XPLMCommandRef xcref = XPLMFindCommand(comm.c_str());
 		if (!xcref)
 		{
 			// COMM does not exist
-			Log::FormatLine(LOG_ERROR, "DMAN", "ERROR: invalid COMM %s", comm.c_str());
+			Log::FormatLine(LOG_ERROR, "DMAN", "ERROR: invalid command %s", comm.c_str());
 			return;
 		}
 
  		XPLMCommandOnce(xcref);
 	}
 
+	void DataManager::BeginCommand(const std::string& comm)
+	{
+		Log::FormatLine(LOG_INFO, "DMAN", "Starting command (value:%s)", comm.c_str());
+
+		XPLMCommandRef xcref = XPLMFindCommand(comm.c_str());
+		if (!xcref)
+		{
+			// Command does not exist
+			Log::FormatLine(LOG_ERROR, "DMAN", "ERROR: invalid command %s", comm.c_str());
+			return;
+		}
+
+		// Flag the command as having been explicitly started by the plugin
+		commandStarted[xcref] = true;
```

Comment:
Rather than a `map`, I think a `std::set` would work well here. Instead of setting true/false, we would do `commandStarted.insert(xcref)` here, and `commandStarted.erase(xcref)` in `EndCommand`.

Reply:
Totally agree, good callout - changed this to an std::set.